packages/nimble-components/src/rich-text/editor/tests/rich-text-editor-mention.spec.ts

```diff
@@ -0,0 +1,831 @@
+        });
+
+        it('updating valid `key` in mapping mention should update it to a mention view if is a absolute link before', async () => {
+            element.setMarkdown('<user:1> <test:2>');
+            await appendUserMentionConfiguration(
+                element,
+                ['user:1'],
+                ['username1']
+            );
+            await appendTestMentionConfiguration(
+                element,
+                ['invalid'],
+                ['testname2']
+            );
+
+            expect(pageObject.getEditorTagNames()).toEqual(['P', 'A', 'A']);
+            expect(pageObject.getEditorLeafContents()).toEqual([
+                'user:1',
+                'test:2'
+            ]);
+
+            const renderedUserMention = element.lastElementChild as RichTextMentionTest;
+            const renderedMappingUser = renderedUserMention.firstElementChild as MappingUser;
+            renderedMappingUser.key = 'test:2';
+            await waitForUpdatesAsync();
+
+            expect(pageObject.getMarkdownRenderedTagNames()).toEqual([
+                'P',
+                `${richTextMentionUsersViewTag}`.toUpperCase(),
+                `${richTextMentionUsersViewTag}`.toUpperCase(),
+                'BR'
+            ]);
+            expect(
+                pageObject.getEditorMentionViewAttributeValues('mention-label')
+            ).toEqual(['username1', 'testname2']);
+        });
+
+        it('updating valid `key` in mapping mention should update it to absolute link if it is a mention before', async () => {
+            element.setMarkdown('<user:1> <test:2>');
+            await appendUserMentionConfiguration(
+                element,
+                ['user:1'],
+                ['username1']
+            );
+            await appendTestMentionConfiguration(
+                element,
+                ['test:2'],
+                ['testname2']
+            );
+
+            expect(pageObject.getMarkdownRenderedTagNames()).toEqual([
+                'P',
+                `${richTextMentionUsersViewTag}`.toUpperCase(),
+                `${richTextMentionUsersViewTag}`.toUpperCase(),
+                'BR'
+            ]);
+            expect(
+                pageObject.getEditorMentionViewAttributeValues('mention-label')
+            ).toEqual(['username1', 'testname2']);
+
+            const renderedUserMention = element.firstElementChild as RichTextMentionTest;
+            const renderedMappingUser = renderedUserMention.firstElementChild as MappingUser;
+            renderedMappingUser.key = 'invalid';
+            await waitForUpdatesAsync();
+
+            expect(pageObject.getEditorTagNames()).toEqual(['P', 'A', 'A']);
+            expect(pageObject.getEditorLeafContents()).toEqual([
+                'user:1',
+                'test:2'
+            ]);
+        });
+    });
+
+    it('should fire "mention" event from configuration elment when there is @mention in editor', async () => {
```

Comment:
We should have some more coverage to verify these cases:

1. `mention-update` is fired when appending to an existing mention
2. `mention-update` is fired when deleting from an existing mention
3. `mention-update` isn't fired when deleting the `@` of an existing mention
4. `mention-update` isn't fired when pasting a block of text that contains a mention in the middle
5. `mention-update` isn't fired for other interactions. For example, typing text near an existing mention (but not part of that mention)

Could you also update the names of these tests to use the actual name of the event (`mention-update`, not `mention`)?

Reply:
Added test cases for these behaviours